client/subscription-product-onboarding/modal.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+import { Button, Icon, Modal } from '@wordpress/components';
+import { __experimentalCreateInterpolateElement as createInterpolateElement } from 'wordpress-element';
+import { useEffect, useState } from '@wordpress/element';
+import { registerPlugin } from '@wordpress/plugins';
+import { removeQueryArgs } from '@wordpress/url';
+import { __ } from '@wordpress/i18n';
+
+import './style.scss';
+
+const { connectUrl } = window.wcpaySubscriptionProductOnboardingModal;
+
+const FinishSetupButton = () => {
+	const [ isFinishingSetup, setIsFinishingSetup ] = useState( false );
+
+	return (
+		<Button
+			disabled={ isFinishingSetup }
+			href={ connectUrl }
+			isBusy={ isFinishingSetup }
+			isPrimary
+			onClick={ () => setIsFinishingSetup( true ) }
+		>
+			{ __( 'Finish setup', 'woocommerce-payments' ) }
+		</Button>
+	);
+};
+
+// TODO: a11y.
+const SubscriptionProductOnboardingModal = () => {
+	const [ isOpen, setOpen ] = useState( true );
+
+	useEffect( () => {
+		if ( window?.history ) {
+			window.history.replaceState(
+				null,
+				null,
+				removeQueryArgs(
+					window.location.href,
+					'wcpay-subscription-saved-as-draft'
+				)
+			);
+		}
+	}, [] );
```

Comment:
>**Question**: Just so I understand this, this just removes the `wcpay-subscription-saved-as-draft` query arg from the URL when the modal is rendered right?

Yep, that's correct.

>Is that to remove from the browser history the arg that is used to display the modal so pressing back doesn't redisplay it or is there some other intent?
Yep, also to prevent any unintended side-effects from refreshing the page with that query param still present.